ui/apps/platform/src/Containers/Clusters/cluster.helpers.tsx

```diff
@@ -302,26 +303,19 @@ export function formatBuildDate(orchestratorMetadata) {
         : 'Not available';
 }
 
-type ProviderMetadata = {
-    region: string;
-    aws?: any;
-    azure?: any;
-    google?: any;
-};
-
-export function formatCloudProvider(providerMetadata: ProviderMetadata) {
+export function formatCloudProvider(providerMetadata: ClusterProviderMetadata) {
     if (providerMetadata) {
         const { region } = providerMetadata;
 
-        if (providerMetadata.aws) {
+        if ('aws' in providerMetadata) {
```

Comment:
Is this a TS thing, or just a preference?

Reply:
Good question. TypeScript did not distinguish the types with property dot but did with `in` operator.

TypeScript docs site is sadly not very searchable to find an example, let alone a guideline about it.

Reply:
By the way, the original `if` statements are my preference.
* TypeScript does understand that idiom for optional properties with `?` or `Whatever | null` types.
* My guess is that TypeScript prefers `in` operator as an explicit attempt to discriminate the type union.